crates/dojo-lang/src/compiler.rs

```diff
@@ -207,41 +210,50 @@ fn update_manifest(
     manifest: &mut dojo_world::manifest::Manifest,
     db: &RootDatabase,
     crate_ids: &[CrateId],
-    compiled_artifacts: HashMap<SmolStr, (FieldElement, Option<abi::Contract>)>,
+    compiled_artifacts: HashMap<SmolStr, (FieldElement, Option<abi::Contract>, Vec<BigUintAsHex>)>,
 ) -> anyhow::Result<()> {
     fn get_compiled_artifact_from_map<'a>(
-        artifacts: &'a HashMap<SmolStr, (FieldElement, Option<abi::Contract>)>,
+        artifacts: &'a HashMap<SmolStr, (FieldElement, Option<abi::Contract>, Vec<BigUintAsHex>)>,
         artifact_name: &str,
-    ) -> anyhow::Result<&'a (FieldElement, Option<abi::Contract>)> {
+    ) -> anyhow::Result<&'a (FieldElement, Option<abi::Contract>, Vec<BigUintAsHex>)> {
         artifacts.get(artifact_name).context(format!(
             "Contract `{artifact_name}` not found. Did you include `dojo` as a dependency?",
         ))
     }
 
     let world = {
-        let (hash, abi) = get_compiled_artifact_from_map(&compiled_artifacts, WORLD_CONTRACT_NAME)?;
+        let (hash, abi, source) =
+            get_compiled_artifact_from_map(&compiled_artifacts, WORLD_CONTRACT_NAME)?;
         Contract {
             name: WORLD_CONTRACT_NAME.into(),
             abi: abi.clone(),
+            source: source.clone(),
```

Comment:
Is this the sierra? Is it enough to verify a contract + abi with just sierra? I think we want to store the cairo code itself, so we can compile it down to sierra -> casm to verify?

Reply:
You are right, sierra code must be compiled by the verifier

Reply:
@tarrencev I am a bit confused, about what we expect:
- Everything in the manifest
- Everything in separate files abi + source per contract
- Everything in separate files (1 file for abi + 1 file for the source per contract)

Reply:
Since we want them to be uploaded into IPFS, I assume it would be better with 2 files per contract, right?

Reply:
@Bal7hazar I think the 3rd option would be best, with the flattened source code. I'm not sure exactly how to extract that from the compiler but it should be possible. Ideally we could support the file structure as well, but it might be tricky. Perhaps we handle source in a separate commit and start with abi?

Reply:
>  Perhaps we handle source in a separate commit and start with abi?

@tarrencev, actually I am not sur the abi is required for a verification purpose. If we can have the cairo sources, we should be able to generate abi from it, right?

Reply:
@Bal7hazar yes that is true but abi can still be useful for interacting with the contracts without verification

Reply:
I have overspent time on it, and it is still confused to me.
I think it will be more efficient if someone else knowing better dojo internal flows and rust could work on it
I can prefer to pause this PR, leaving it for anyone wanting to continue or restart from the beginning.